assets/js/private-articles.tsx

```diff
@@ -0,0 +1,233 @@
+import { h, render, Fragment, createContext, ComponentChildren } from "preact";
+import { useEffect, useMemo, useContext } from "preact/hooks";
+import { load as loadYAML } from "js-yaml";
+import { marked } from "marked";
+import Markup from "preact-markup";
+import { createPortal } from "preact/compat";
+import { useQuery } from "preact-fetching";
+import { useCurrentUser, PrivateLoginLink } from "./private";
+import { Alert } from "./components/alert";
+import { fetchGitHubFile } from "./github";
+import { Navigation, SidenavWithWrapper } from "./components/sidenav";
+import { AnchorLink } from "./components/anchor-link";
+
+const GitHubContext = createContext({
+  ref: undefined,
+} as { ref?: string });
+
+export function PrivateArticlesIndex() {
+  const nav = document.getElementById("sidenav-wrapper") as HTMLElement;
+  const [currentUser] = useCurrentUser();
+  const { ref } = useContext(GitHubContext);
+
+  const { data: articles } = useQuery(`articlesIndex`, () => {
+    if (!currentUser) {
+      return [];
+    }
+    return fetchGitHubFile({
+      token: currentUser.token,
+      repo: "18f/identity-handbook-private",
+      path: "_articles",
+      ref,
+    }).then((dir) => {
+      if (!Array.isArray(dir)) {
+        return [];
+      }
+
+      return dir;
+    });
+  });
+
+  return (
+    <>
+      <ul>
+        {articles &&
+          articles.map((article) => (
+            <li key={article.path}>
+              <a
+                href={`?article=${encodeURIComponent(article.path)}`}
+                className="usa-link"
+              >
+                {article.name}
+              </a>
+            </li>
+          ))}
+      </ul>
+      {createPortal(<SidenavWithWrapper navigation={[]} />, nav)}
+    </>
+  );
+}
+
+interface Frontmatter {
+  title: string;
+  description?: string;
+  category: string;
+}
+
+interface Heading {
+  text: string;
+  depth: number;
+}
+
+function buildNavigation(headings: Heading[]): Navigation {
+  const navigation: Navigation = [];
+  let lastDepth = 0;
+  const stack: Navigation[] = [];
+  headings.forEach(({ text, depth }) => {
+    if (!lastDepth) {
+      navigation.push(text);
+      stack.push(navigation);
+    } else if (depth > lastDepth) {
+      const last = stack[stack.length - 1];
+      const nested = [text];
+      last.push(nested);
+      stack.push(nested);
+    } else if (depth === lastDepth) {
+      const last = stack[stack.length - 1];
+      last.push(text);
+    } else {
+      while (depth < lastDepth) {
+        stack.pop();
+        lastDepth -= 1;
+      }
+      const last = stack[stack.length - 1];
+      last.push(text);
+    }
+
+    lastDepth = depth;
+  });
+  return navigation;
+}
+
+function buildHeader(Tag: any) {
```

Comment:
Could we type this to a string literal union?

```suggestion
function buildHeader(Tag: "h1" | "h2" | "h3" | "h4" | "h5" | "h6") {
```

Or:

```suggestion
type HeadingTag = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";

function buildHeader(Tag: HeadingTag) {
```